raku-advent-2020/local-drafts/p6steve-rudolph-on-raku.md

```diff
@@ -0,0 +1,183 @@
+Now he knows where he is, Rudolph can set a course to steer home to the North Pole. But wait, how can
+he adjust for the difference between Magnetic north on his Compass and True North, his destination?
+
+Rudolph has another trick up his (antler) sleeve:
+```
+class CompassAdjustment { ... }
+
+our $variation = 0;			#optional variation (Compass-Adjustment)
+our $deviation = 0;			#optional deviation (Compass-Adjustment)
+
+#| Bearing embodies the identity 'M = T + Vw', so...
+#| Magnetic = True + Variation-West [+ Deviation-West]
+class Bearing is NavAngle is export {
+	has Real  $.value is rw where 0 <= * <= 360; 
+	has Str   $.compass where <M T>.any;
+
+	method M {
+		if $.compass eq <M> { return( self ) } 
+		else { return( self + ( $variation + $deviation ) ) }
+	}
+	method T {
+		if $.compass eq <T> { return( self ) } 
+		else { return( self - ( $variation + $deviation ) ) }
+	}
+
+   sub check-same( $l, $r ) {
+	if $r ~~ CompassAdjustment { 
+		return 
+	}
+        if ! $l.compass eq $r.compass {
+            die "Cannot combine Bearings of different Types!"
+        }    
+    }  
+    method add( $r is rw ) {
+        my $l = self;
+        check-same( $l, $r );
+        $l.value += $r.value;
+ 	$l.compass( $r.compass );
+        return $l
+    }    
+    method subtract( $r is rw ) {
+        my $l = self;
+        check-same( $l, $r );
+        $l.value -= $r.value; 
+	$l.compass( $r.compass );
+        return $l
+    }
+}
+
+class CompassAdjustment is Bearing is export {
+	has Real  $.value is rw where -180 <= * <= 180; 
+
+	multi method compass {						#get compass
+		given $.value {
+			when * >= 0 { return <W>, 0 }
+			when * < 0  { return <E>, 1 }
+		}
+	}
+	multi method compass( Str $compass ) {				#set compass
+		given $compass {
+			when <W>   { }		#no-op
+			when <E>   { $.value = -$.value }
+			default    { die "Compass-Adjustment must be <W E>.any" }
+		}
+	}
+}
+```
+Now, after setting the compass variation, Rudolph can enter in their magnetic compass reading and get back the 
+Bearing to True North.
+```
+$Physics::Navigation::variation = CompassAdjustment.new( value => 7, compass => <W> );
+
+my $bear ♓️ <43°30′30″M>; say ~$bear;   #OUTPUT 43°30.5 M
+say ~$bear.T;				#OUTPUT 43°37.5 T
+```
+Santa can even steer (H=Helm) by doing addition/subtraction of the course change Bearing since +/- are already overidden
+for Physics::Measure objects - that's one benefit of these weird ♓️ unicode operators ... they act as a warning that 
+language mutations are active in these code regions.
+
+And should Santa be bringing home a sleigh full unwanted ferrous Christmas presents (bikes, climbing frames, 
+meccano sets and so on), then this can be accommodated with the ```$Physics::Navigation::deviation``` setting.
+
+And finally Santa, Rudolph and the other reindeer can rest their weary bones around the glowing fire at home
+after a long night's work!
+
+Merry Christmas
+p6steve.   (p6 is pronounced "Physics")
```

Comment:
He, I didn't now that :-) Nice pun.

Reply:
Yes - I even voted for and now prefer 'raku' though!